web_validation_dialog/static/src/js/web_validation_dialog.js

```diff
@@ -0,0 +1,114 @@
+                            click: function() {
+                                this.close();
+                                self.open_pincode_dialog();
+                            },
+                        },
+                        {
+                            text: _t('Cancel'),
+                            close: true
+                        },
+                    ]
+                });
+            } else if (self.node.attrs.confirm) {
+                    var def = $.Deferred();
+                    Dialog.confirm(self, self.node.attrs.confirm, {confirm_callback: self.on_confirmed}).
+                        on("closed", null, function() {
+                            def.resolve();
+                        });
+                    return def.promise();
+                } else {
+                    return self.on_confirmed();
+                }
+        },
+        execute_action: function() {
+            var self = this;
+            if (!this.node.attrs.special) {
+                return this.view.recursive_save().then(self.exec_action());
+            } else {
+                return self.exec_action();
+            }
+        },
+        open_pincode_dialog : function() {
+            var self = this;
+            new Dialog(self, {
+                title: _t('Validation'),
+                size : "small",
+                $content: QWeb.render('DialogValidation'),
+                buttons: [
+                            {
+                                text: _t("Ok"),
+                                classes: 'btn-primary',
+                                click: function() {
+                                   var curr_obj = this;
+                                   var password = this.$el.find("#pincode").val();
+                                   if (password) {
+                                       framework.blockUI();
+                                       var callback = self.validate_pincode(self.is_dialog_security, password);
+                                       callback.done(function(result) {
+                                           framework.unblockUI();
+                                           if (result) {
+                                               curr_obj.close();
+                                              self.on_confirmed();
+                                           } else {
+                                               Dialog.alert(self, _t("Invalid or Wrong Password! Contact your Administrator."));
+                                               return;
+                                           }
+                                       }).fail(function(error) {
+                                           framework.unblockUI();
+                                           Dialog.alert(self, _t("Either the password is wrong or the connection is lost! Contact your Administrator."));
+                                           return;
```

Comment:
Is this return really needed? as well as the other returns in the click cb function?

Reply:
@tarteo I think is not needed but if open Dialog alert box then return something(is good for us). so we added return.

Reply:
You don't return anything. So the way you use `return;` (on this line and line 83 and 92) it does the same as no return statement. Empty returns are mostly used to 'escape' execution of the rest of the function.